rpc/lib/types/types_test.go

```diff
@@ -19,20 +19,41 @@ func TestResponses(t *testing.T) {
 	assert := assert.New(t)
 	cdc := amino.NewCodec()
 
-	a := NewRPCSuccessResponse(cdc, "1", &SampleResult{"hello"})
+	a := NewRPCSuccessResponse(cdc, JSONRPCStringID("1"), &SampleResult{"hello"})
 	b, _ := json.Marshal(a)
 	s := `{"jsonrpc":"2.0","id":"1","result":{"Value":"hello"}}`
 	assert.Equal(string(s), string(b))
 
-	d := RPCParseError("1", errors.New("Hello world"))
+	d := RPCParseError(JSONRPCStringID("1"), errors.New("Hello world"))
 	e, _ := json.Marshal(d)
 	f := `{"jsonrpc":"2.0","id":"1","error":{"code":-32700,"message":"Parse error. Invalid JSON","data":"Hello world"}}`
 	assert.Equal(string(f), string(e))
 
-	g := RPCMethodNotFoundError("2")
+	g := RPCMethodNotFoundError(JSONRPCStringID("2"))
 	h, _ := json.Marshal(g)
 	i := `{"jsonrpc":"2.0","id":"2","error":{"code":-32601,"message":"Method not found"}}`
 	assert.Equal(string(h), string(i))
+
+	j := NewRPCSuccessResponse(cdc, JSONRPCIntID(1), &SampleResult{"hello"})
+	k, _ := json.Marshal(j)
+	l := `{"jsonrpc":"2.0","id":1,"result":{"Value":"hello"}}`
+	assert.Equal(string(l), string(k))
+}
+
+func TestUnmarshallResponses(t *testing.T) {
+	assert := assert.New(t)
+	cdc := amino.NewCodec()
+	var err error
+	response := &RPCResponse{}
+	err = json.Unmarshal([]byte(`{"jsonrpc":"2.0","id":"1","result":{"Value":"hello"}}`), response)
```

Comment:
Mind writing a [table driven test](https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go) for this? We can try multiple strings ("", "1", "alphabet", something with non-ascii, etc.), multiple ints (-1, 0, 1, 100, etc.), and ensure we get failures for things that aren't strings or ints. Technically it should allow negative ints as IDs as well (according to the JSON spec for "Numbers").

Can we also add tests to the client and/or server ensuring that we get Responses with the same ID as sent in the Request, both for strings and ints?